### PR TITLE
feat: lock client

### PR DESCRIPTION
**Pull request by LingKa28**
Base on #7 #8  

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Add new feature.

* what changes does this pull request make?
  * add lock client.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: tests/lock_test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    await lock_client.lock(b"lock-test")
+
+    try:
+        await asyncio.wait_for(lock_client.lock(b"lock-test"), timeout=1)
+        assert False
+    except asyncio.TimeoutError:
+        pass
+
+    try:
+        await asyncio.wait_for(lock_client.lock(b"lock-test"), timeout=0.6)
```
Comment (Phoenix500526) [inline tests/lock_test.py:39]: This test case implies the condition that the lock will time out after 1 second. Please simplify this test case and make it clearer.

File: tests/lock_test.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    except asyncio.TimeoutError:
+        assert False
+
+    await lock_client.unlock(b"lock-test")
+
+@pytest.mark.asyncio
+async def test_lock_should_unlock_after_cancelled():
```
Comment (Phoenix500526) [inline tests/lock_test.py:71]: Please modify the test case name to make it clearer.

File: tests/lock_test.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    await lock_client.unlock(b"lock-test")
+
+    try:
+        await asyncio.wait_for(lock_client.lock(b"lock-test"), timeout=0.2)
+    except asyncio.TimeoutError:
+        assert False
+
+
+
+# class TestLockClient(unittest.TestCase):
```
Comment (Phoenix500526) [inline tests/lock_test.py:90]: If we don't need this `TestLockClient` anymore, please remove it.

File: tests/lock_test.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+from client import client
+
+@pytest.mark.asyncio
+async def test_lock_unlock_should_success_in_normal_path():
+    """
+    Test lock unlock should success in normal path
+    """
+    curp_members = ["172.20.0.3:2379", "172.20.0.4:2379", "172.20.0.5:2379"]
+    cli = await client.Client.connect(curp_members)
+    lock_client = cli.lock_client
```
Comment (Phoenix500526) [inline tests/lock_test.py:14]: Maybe we can use a test fixture to make this test case more brief. FYI: https://stackoverflow.com/questions/49936724/async-fixtures-with-pytest

File: client/watch.py [hunk 5]
```diff
@@ -43,7 +45,7 @@ async def watch():
             yield WatchRequest(create_request=req)
 
             while not self.is_cancel:
-                await asyncio.sleep(0.5)
+                await asyncio.sleep(0.2)
```
Comment (Phoenix500526) [inline client/watch.py:48]: How did we arrive at this 0.2?

File: tests/lock_test.py (hunk 1 above)
Comment (LingKa28) [reply tests/lock_test.py]: Done.

File: tests/lock_test.py (hunk 3 above)
Comment (LingKa28) [reply tests/lock_test.py]: Done.

File: client/watch.py (hunk 5 above)
Comment (LingKa28) [reply client/watch.py]: I accidentally changed it during testing, and in fact we don't need to change it.

File: tests/lock_test.py (hunk 2 above)
Comment (LingKa28) [reply tests/lock_test.py]: Done.

File: tests/lock_test.py (hunk 4 above)
Comment (LingKa28) [reply tests/lock_test.py:15]: We will try to use this in after pr.

File: tests/lock_test.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    await lock_client.lock(b"lock-test", ttl=2)
+
+    try:
+        await asyncio.wait_for(lock_client.lock(b"lock-test"), timeout=2)
+        msg = "Lock should timeout when ttl is set"
+        raise AssertionError(msg)
+    except asyncio.TimeoutError:
+        pass
+
```
Comment (Phoenix500526) [inline tests/lock_test.py:61]: You should try to lock here to prove that the lock will release after ttl.

Comment (LingKa28) [reply tests/lock_test.py:61]: Done.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by themanforfree — APPROVED** [review]

**Pull request merged**